specifications/client/verifying_client.md

```diff
@@ -0,0 +1,125 @@
+# Diem Verifying Client Spec
+
+THIS IS A WORK IN PROGRESS
+
+Links:
```

Comment:
can you add a few lines on what a verifying client is? And how it contrasts with a non-verifying client? Possibly linking to the non-verifying client specs